src/components/TodoCreationForm.js

```diff
@@ -1,27 +1,52 @@
 import React from "react";
+import DatePicker from "react-datepicker";
+import Button from "./Library/Button";
 
 export default class TodoCreationForm extends React.Component {
   state = {
-    taskName: ""
+    todoName: "",
+    dueDate: new Date()
   }
 
+  handleDueDateChange = date => {
+    this.setState({
+      dueDate: date
+    });
+  };
+
   render() {
     return (
       <form
+        className="todo-creation-form"
         onSubmit={e => {
           e.preventDefault();
-          if (this.state.taskName === "") return;
-          this.props.onCreate(this.state.taskName)
-          this.setState({taskName: ""})
+          if (this.state.todoName === "") return;
+          this.setState({todoName: ""})
         }}
       >
         <input
           className="input is-primary is-small"
           autoFocus
-          placeholder="Add a task (hit Enter to validate)"
-          value={this.state.taskName}
-          onChange={e => this.setState({taskName: e.target.value})}
+          placeholder="Add a todo (hit Enter to validate)"
+          value={this.state.todoName}
+          onChange={e => this.setState({todoName: e.target.value})}
+        />
+        <DatePicker
+          selected={this.state.dueDate}
+          onChange={this.handleDueDateChange}
```

Comment:
As we do nothing specifically complicated, why not directly inline the `setState` thing here like we did for the text input ? 
`onChange={newDate => this.setState({ dueDate: newDate })}`
Generally, I tend to inline the callback action when it has only one instruction, maximum two. In this case there is only one instruction: `this.setState({ dueDate: newDate })`

Reply:
Not sure I understood but I did something LOL